src/fontra/client/web-components/menu-panel.js

```diff
@@ -197,6 +201,36 @@ class MenuPanel extends SimpleElement {
     }
   }
 
+  searchMenuItems(key) {
+    // Returns true only for letters, numbers & spaces
+    const isValidSearchInput = /^[a-zA-Z0-9 ]$/.test(key);
+    if (isValidSearchInput) {
```

Comment:
Suggest to return immediately:
```js
    if (!isValidSearchInput) {
      return;
    }
```

(This reduces the indentation for the rest of the function, which in my book is always a win.)